pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -537,6 +537,9 @@ export async function resolveDependencies (
       newPreferredVersions[resolvedPackage.name] = {}
     }
     if (!newPreferredVersions[resolvedPackage.name][resolvedPackage.version]) {
+      if (newPreferredVersions[resolvedPackage.name] === preferredVersions[resolvedPackage.name]) {
+        newPreferredVersions[resolvedPackage.name] = { ...preferredVersions[resolvedPackage.name] }
+      }
```

Comment:
alternative version:
```
    if (!Object.prototype.hasOwnProperty.call(newPreferredVersions, resolvedPackage.name)) {
      newPreferredVersions[resolvedPackage.name] = Object.create(preferredVersions[resolvedPackage.name])
    }
```

let me check if there is a difference in performance for this